### PR TITLE
K8SPSMDB-514 fix cronJob resources

### PR DESCRIPTION
**Pull request by hors**
[K8SPSMDB-514](https://jira.percona.com/browse/K8SPSMDB-514) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/727/commits/6403693b2d64c4fd3e87807b2d8bb14c14d5cb70
image: `perconalab/percona-server-mongodb-operator:PR-727-6403693b`

File: pkg/psmdb/backup/job.go [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+	"github.com/percona/percona-server-mongodb-operator/pkg/psmdb"
 )
 
-func BackupCronJob(backup *api.BackupTaskSpec, crName, namespace string, backupSpec api.BackupSpec, imagePullSecrets []corev1.LocalObjectReference) *batchv1b.CronJob {
+func BackupCronJob(backup *api.BackupTaskSpec, crName, namespace string, backupSpec api.BackupSpec, imagePullSecrets []corev1.LocalObjectReference) (batchv1b.CronJob, error) {
 	jobName := crName + "-backup-" + backup.Name
-
+	resources, err := psmdb.CreateResources(backupSpec.Resources)
+	if err != nil {
+		return batchv1b.CronJob{}, fmt.Errorf("cannot parse Backup resources: %w", err)
```
Comment (egegunes) [inline pkg/psmdb/backup/job.go:19]: please use github.com/pkg/errors package

File: pkg/controller/perconaservermongodb/backup.go [hunk 2]
```diff
@@ -22,17 +22,20 @@ func (r *ReconcilePerconaServerMongoDB) reconcileBackupTasks(cr *api.PerconaServ
 	ls := backup.NewBackupCronJobLabels(cr.Name)
 
 	for _, task := range cr.Spec.Backup.Tasks {
-		cjob := backup.BackupCronJob(&task, cr.Name, cr.Namespace, cr.Spec.Backup, cr.Spec.ImagePullSecrets)
+		cjob, err := backup.BackupCronJob(&task, cr.Name, cr.Namespace, cr.Spec.Backup, cr.Spec.ImagePullSecrets)
+		if err != nil {
+			return errors.Wrap(err, "can't createjob")
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/backup.go:27]:
```suggestion
			return errors.Wrap(err, "can't create job")
```

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

File: pkg/psmdb/backup/job.go (hunk 1 above)
Comment (hors) [reply pkg/psmdb/backup/job.go]: @egegunes  it was done 44f5c08

**Review by egegunes — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| upgrade | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/727/commits/44f5c08d29c07722d3ccacc3c48658f81164c8cd
image: `perconalab/percona-server-mongodb-operator:PR-727-44f5c08d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/727/commits/bf5cfe5bbe10d79471b11d399efbb6d2aa6a35b7
image: `perconalab/percona-server-mongodb-operator:PR-727-bf5cfe5b`

**Review by qJkee — APPROVED** [review]

**Pull request merged**